spec/models/user_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Evans') }
+  before { subject.save }
+
+  it 'name must not be blank' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'posts_counter must be an integer' do
+    subject.posts_counter = 'rice'
+    expect(subject).to_not be_valid
+  end
+
+  it 'posts_counter should be greater than or equal to 0' do
+    subject.posts_counter = 0
+    subject.posts_counter = -5
+    expect(subject).to_not be_valid
+  end
+
+  it 'returns no more than 3 posts' do
+    expect(subject.three_recent_posts.length).to be <= 3
+  end
```

Comment:
- [ ] Well! Good job so far! I noticed that there might be a misunderstanding of the previous reviewer's comments. Additionally, you deleted the existing test cases and added new ones, which is not the recommended practice. As the previous review suggests, the intention is not only to check the correct number of posts but also whether the correct post is returned.

Here's an example of how you can modify the test case:

```ruby
it 'returns no more than 3 posts' do
  user = User.create(name: 'name', photo: 'url', bio: 'bio', posts_counter: 0)
  post1 = Post.create(author_id: user.id, title: 'post', text: 'text', likes_counter: 0, comments_counter: 0,
                      created_at: 4.days.ago)
  post2 = Post.create(author_id: user.id, title: 'post', text: 'text', likes_counter: 0, comments_counter: 0,
                      created_at: 3.days.ago)
  post3 = Post.create(author_id: user.id, title: 'post', text: 'text', likes_counter: 0, comments_counter: 0,
                      created_at: 2.days.ago)
  post4 = Post.create(author_id: user.id, title: 'post', text: 'text', likes_counter: 0, comments_counter: 0,
                      created_at: 1.day.ago)
  user.reload
  recent_posts = user.three_recent_posts
  expect(recent_posts.length).to be <= 3
  expect(recent_posts).to include(post2)
  expect(recent_posts).to include(post3)
  expect(recent_posts).to include(post4)
end
```

- [ ] Please note that the code provided may not be directly applicable to your codebase as it was not tested due to migration issues. However, it can serve as a useful reference. Please review all comments from the previous reviewer and update your code accordingly.

This approach ensures that the existing test cases are not deleted and new assertions are added. Keep up the good work! 🚀👩‍💻